### PR TITLE
Add support for kubevirt platform

### PR DESCRIPTION
**Pull request by ravidbro**
Signed-off-by: Ravid Brown <ravid@redhat.com>

### CODE REVIEW

File: pkg/operator/genericoperatorclient/dynamic_operator_client_test.go [hunk 1]
```diff
@@ -33,7 +33,6 @@ func TestSetOperatorSpecFromUnstructured(t *testing.T) {
 					"non-standard-field":         "value",
 					"logLevel":                   "Trace",
 					"managementState":            "",
-					"operatorLogLevel":           "",
```
Comment (sttts) [inline pkg/operator/genericoperatorclient/dynamic_operator_client_test.go:36]: these changes look unrelated. If necessary, put into extra commit and describe why youneed them.

**Comment by sttts** [comment]
Looks nearly fine, only https://github.com/openshift/library-go/pull/915#discussion_r503169668.

File: pkg/operator/genericoperatorclient/dynamic_operator_client_test.go (hunk 1 above)
Comment (ravidbro) [reply pkg/operator/genericoperatorclient/dynamic_operator_client_test.go:36]: It is related and a must, the test fails without this change.
The new version of 'api' changes the behavior of the response and that what is expected from now.

Comment (sttts) [reply pkg/operator/genericoperatorclient/dynamic_operator_client_test.go:36]: Then move it into its own commit.

Also please don't mix vendor bump with other changes.

Comment (ravidbro) [reply pkg/operator/genericoperatorclient/dynamic_operator_client_test.go:36]: done

Comment (ravidbro) [reply pkg/operator/genericoperatorclient/dynamic_operator_client_test.go:36]: I reorganized the commits in this PR
